contracts/earn/strategies/Geist/StrategyGeist.sol

```diff
@@ -0,0 +1,288 @@
+        }
+    }
+
+    /* ----- Private Functions ----- */
+
+    function _swapRewards(uint256 reserveWantGTokenBal) private {
+        // reward to wNative
+        uint256 rewardBal = IERC20(reward).balanceOf(address(this));
+        IUniswapRouter(uniRouter).swapExactTokensForTokens(rewardBal, 0, rewardToWNativeRoute, address(this), now);
+
+        for (uint256 i; i < rewards.length; i++) {
+            uint256 gTokenToWithdraw = IERC20(rewards[i].gToken).balanceOf(address(this));
+
+            // if reward is wantToken, we have to substrate the reserved gToken balance.
+            if (rewards[i].gToken == wantToken.gToken) {
+                gTokenToWithdraw = gTokenToWithdraw.sub(reserveWantGTokenBal);
+            }
+
+            if (gTokenToWithdraw > 0) {
+                // gToken to the underlying asset
+                ILendingPool(lendingPool).withdraw(rewards[i].token, gTokenToWithdraw, address(this));
+
+                uint256 tokenToSwap = IERC20(rewards[i].token).balanceOf(address(this));
+                if (rewards[i].token != wNative && rewards[i].token != wantToken.token) {
+                    IUniswapRouter(uniRouter).swapExactTokensForTokens(
+                        tokenToSwap,
+                        0,
+                        extraRewardToWNativeRoutes[i],
+                        address(this),
+                        now
+                    );
+                }
+            }
+        }
+
+        uint256 wNativeBal = IERC20(wNative).balanceOf(address(this));
+        if (wNative != want && wNativeBal > 0) {
+            IUniswapRouter(uniRouter).swapExactTokensForTokens(wNativeBal, 0, wNativeToWantRoute, address(this), now);
+        }
+    }
+
+    /* ----- Admin Functions ----- */
+
+    function addRewardToNativeRoute(address[] memory _rewardToWNativeRoute) external onlyAdmin {
+        address _token = _rewardToWNativeRoute[0];
+        (address _gToken, , ) = IDataProvider(dataProvider).getReserveTokensAddresses(_token);
+
+        rewards.push(TokenAddresses(_token, _gToken));
+        extraRewardToWNativeRoutes.push(_rewardToWNativeRoute);
+
+        TransferHelper.safeApprove(_token, uniRouter, uint256(-1));
+    }
+
+    function removeRewardToWNativeRoute() external onlyAdmin {
+        IERC20(rewards[rewards.length - 1].token).safeApprove(uniRouter, 0);
+
+        rewards.pop();
+        extraRewardToWNativeRoutes.pop();
+    }
+
+    function retireStrat() external override {
+        require(msg.sender == vault, 'StrategyGeist: !vault');
+        ILendingPool(lendingPool).withdraw(want, type(uint256).max, address(this));
+        uint256 wantBal = IERC20(want).balanceOf(address(this));
+        if (wantBal > 0) {
+            TransferHelper.safeTransfer(want, vault, wantBal);
+        }
+    }
+
+    function emergencyExit() external override onlyAdmin {
+        ILendingPool(lendingPool).withdraw(want, type(uint256).max, address(this));
```

Comment:
type(uint256).max

Please confirm it works in testing.

Reply:
sure